src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -399,9 +399,12 @@ public void onClick(ClickEvent event)
       sourceButton_.setTitle(SOURCE_BUTTON_TITLE);
       toolbar.addRightWidget(sourceButton_);
 
-      previewButton_ = commands_.previewJS().createToolbarButton(false);
-      toolbar.addRightWidget(previewButton_);
+      previewJsButton_ = commands_.previewJS().createToolbarButton(false);
+      toolbar.addRightWidget(previewJsButton_);
       
+      previewSqlButton_ = commands_.previewSql().createToolbarButton(false);
```

Comment:
this probably needs an icon that doesn't have the globe on it (which implies HTML) -- maybe the green preview arrow over an icon of a data grid? (we can get this from paul after merge if you like)

Reply:
Yeah, makes sense, I'll ask Paul for icons on SQL preview and D3 preview, but could be refined a bit.